package-system/zlib/FindZLIB_compat_windows.cmake

```diff
@@ -28,4 +28,6 @@ set(ZLIB_PATCH_VERSION "11")
 
 add_library(ZLIB::ZLIB INTERFACE IMPORTED GLOBAL)
 set_target_properties(ZLIB::ZLIB PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${ZLIB_INCLUDE_DIRS}")
-target_link_libraries(ZLIB::ZLIB INTERFACE ${ZLIB_LIBRARIES})
+set_target_properties(ZLIB::ZLIB PROPERTIES IMPORTED_LOCATION "${ZLIB_LIBRARIES}")
```

Comment:
looks like the only difference between these two "compat" files is the name of the lib. Instead of duplicating so much, you can handle it as a "trait" and just have these files have the set for `ZLIB_LIBRARIES`

If these Find files are entry points, they can just include a common file that sets up the rest and just have the set for `ZLIB_LIBRARIES`

Reply:
This is actually the entry point.   And I'm biasing to simplest possible file structure most readable.  There's no reason to include a common file here becuase each platform only gets their own file.  E.g. on mac you only get the one FindFile and its perfect for mac.  Adding a more complicated include heirarchy is possible but what does it buy?

Reply:
The other thing to consider is that zlib being special means I'm probably not going to recommend people copy this.

What I'd much prefer is if we made a canonical version of say, libtiff  and made it behave like the official cmake one, ie, https://cmake.org/cmake/help/latest/module/FindTIFF.html

This means naming our 3p "TIFF" and not "tiff" and doing things in it like 
```
if (some variable that exists if we're within the O3DE ecosystem)
   (use o3de functions and make 3rdParty::TIFF
else()
  (make TIFF::TIFF and make it look like the output from default FindTIFF shipped with cmake
```

which woudl mean only 1 "find file" instead of having to make a 'compat".  unfortunatly the standard is ZLIB and we called ours zlib.

Reply:
Which makes me think, what if I went ahead and updated this now for zlib.  Ie, changed o3de to aslk for 3rdParty::ZLIB instead of 3rdParty::zlib and then tried this?  Its worth a shot.

Reply:
> Adding a more complicated include heirarchy is possible but what does it buy?

Mostly maintenance of these files, if that is a concern, I am ok dropping this. I know we are trying to simplify these scripts, so this was a potential simplification